### ISSUE TITLE
How do you parse fansubdb?

### ISSUE THREAD

**Issue by FansubDB** [qgustavor/fansubdb#4]
Hello

Like my title said, how do you parse fansubdb website to do this [kind of file](https://github.com/qgustavor/fansubdb/blob/master/_data/seasons/2017-outono.yml)?

Thank you in advance

**Comment by qgustavor** [comment]
Just the anime metadata and simulcast data is parsed from FansubDB. I use simple HTML parsing, using [cheerio](https://github.com/cheeriojs/cheerio/), for that.

Well, I know that MediaWiki have API endpoints, [like this](https://fansubdb.com/w/api.php?action=query&titles=Spring_2018&prop=revisions&rvprop=content&format=jsonfm&formatversion=2), but I'm using HTML parsing because I reused data and some code from a research I did and [posted on /r/anime last year](https://reddit.com/r/anime/comments/5luvxz/i_wanted_to_miss_fansubs_too/) (part of this data comes from FansubDB, other part comes from AniDB).

For some reason at this time I used HTML parsing. Maybe I didn't knew about the MediaWiki API. For today as data isn't being fetched automatically I'm still using this old code, but in case I make a tool which updates anime metadata automatically (which would be handy for new seasons, as they're constantly getting new updates) I think that I will use API endpoints. Before I need to think in a way to automatically update this data on YAML files without losing the changes I did manually on those... :expressionless: 

The data about fansubs I'm getting manually by visiting each group website. I made a tool to help me search groups (using Wordpress and Blogger API) and get alerts about new releases and other relevant info (using their feeds), but most of this process is manual.

Translations of common things are handled by some search/replace functions. Sometimes those functions make some mistakes or don't translate everything, so [I need to fix manually](https://github.com/qgustavor/fansubdb/commit/0bafb1395a899eaacad4000ff281e5f6f92e3089).

I get the simulcast data from FansubDB, then I filter those which translate to Portuguese: Crunchyroll, Netflix and HIDIVE. When they translate to English they often translate to Portuguese too.

AniDB links I got by downloading their season pages (manually, using Chrome) then using cheerio to parse it and then matching entries by name. Sometimes names aren't equal, so I use a levenshtein distance function to help. Because of that there are entries on FansubDB which don't have links to AniDB, but here they're linked, hentai entries don't have links (because AniDB hides those on season pages) and there are some entries with mismatched URLs (I plan to fix those later).

Well, in the end not everything is auto-converted, there's a lot of manual work and most of the conversion is handled by a 244 lines Node.js script, with only bare requires, so it's quite messed.
